### ISSUE TITLE
Bugs in the json parser

### ISSUE THREAD

**Issue by Shredded-Pork** [VectorSpaceLab/EditScore#4]
There are many bugs when parsing the outpu of model. For example, 

<img width="566" height="605" alt="Image" src="https://github.com/user-attachments/assets/4723edae-3310-419c-a134-3f37bd7bcfe7" />

which will make the reward model output the repeated result and don't return the score. In the condition, 'give up'  doesn't work. I think the results should be tidied up. There are some logical loopholes and it cannot handle situations outside the boundaries😂.

**Comment by Luciennnnnnn** [comment]
Hi, thank you for bringing this issue to our attention. We have used this code across all our RL experiments and have not observed the “never give up” issue, which could be caused by differences in the data. I’ll investigate this and provide a fix shortly❤️.

**Comment by Luciennnnnnn** [comment]
@Shredded-Pork We’ve just released a temporary fix — please let us know if it works for you or if you run into any other issues 😊

**Comment by Luciennnnnnn** [comment]
The ultimate solution to this issue is to use a stronger base model (better json formatting ability). We’ll soon release the 32B and 72B versions, along with Qwen3-VL variants — hopefully, they’ll perform even better! 🚀

**Comment by Luciennnnnnn** [comment]
> Yes, it's common in my data. It seems like that it lacks a , in the output. By the way, when wiil you release a stronger base model and when wiil you release your training data?

I’ve added an improved JSON fix function — it should now return the correct JSON output for your case.

<img width="2060" height="168" alt="Image" src="https://github.com/user-attachments/assets/ad73249f-4a05-4a07-878d-0f4d12893278" />

<img width="2076" height="206" alt="Image" src="https://github.com/user-attachments/assets/17d1c9d4-5a4b-4b48-8279-4e44c5f65b60" />

**Comment by Luciennnnnnn** [comment]
I’ll close this issue for now. If you encounter any further problems, feel free to reopen it.

By the way, the LoRA weight of EditScore-72B is now available.

**Comment by Luciennnnnnn** [comment]
@akk-123 We launched the Qwen3-VL version last week, using llama-factory for SFT. However, there are currently some [bugs](https://github.com/hiyouga/LLaMA-Factory/issues/9282) related to Qwen3-VL finetuning, so we’re waiting for a fix from the llama-factory side.
